main.go

```diff
@@ -23,17 +25,35 @@ func printUsage() {
 	os.Exit(0)
 }
 
+func timeoutSecondsFromCount(count int) time.Duration {
+	waitOnLastOne := time.Duration(2) * time.Second
+	return time.Duration(count)*time.Second + waitOnLastOne
+
+}
+
 // RunPinger takes an Option object, gets a Dialer, and runs a Pinger against
 // the passed target, for count packets.
 func RunPinger(opt *vpn.Options, target string, count uint32) error {
-	conn := vpn.NewRawDialer(opt)
-	pinger := extras.NewPinger(conn, target, int(count))
-	err := pinger.Run()
+
+	c := int(count)
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutSecondsFromCount(c)+extraTimeoutSeconds)
+	defer cancel()
+
+	rawDialer := vpn.NewRawDialerWithContext(opt, ctx)
+	conn, err := rawDialer.Dial()
 	if err != nil {
-		// TODO identify different errors
-		os.Exit(42)
+		return err
 	}
-	pinger.Stop()
+
+	pinger := ping.New(target, conn)
+	pinger.Count = c
+	pinger.Timeout = timeoutSecondsFromCount(c)
+	err = pinger.Run()
```

Comment:
Idiomatically, if you want to support a context, you should have `pinger.RunContext(ctx)`. If you don't care about the non-context case, otherwise: `pinger.Run(ctx)`.